lib/Service/PdfService.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Richdocuments\Service;
+
+use mikehaertl\pdftk\Pdf;
+use OCP\Files\Node;
+use OCP\IConfig;
+use Psr\Log\LoggerInterface;
+
+class PdfService {
+	public function __construct(
+		private IConfig $config,
+		private LoggerInterface $logger
+	) {
+	}
+
+	public function extractFields(Node $file): array {
+		$dataDir = $this->config->getSystemValue('datadirectory');
+		$filePath = $dataDir . $file->getPath();
```

Comment:
This will not work with object storage or other external storage locations. Rather use something like this:


```suggestion
		$filePath = $file->getStorage()->getLocalFile($file->getInternalPath());
```